File: 2017-grid-game/2017-grid-game.py
class Solution:
    def gridGame(self, grid):
        top = sum(grid[0])
        bottom = 0
        res = float('inf')
        
        for i in range(len(grid[0])):
            top -= grid[0][i]
            res = min(res, max(top, bottom))
            bottom += grid[1][i]
        
        return res

File: 2017-grid-game/2017-grid-game.java
class Solution {
    public long gridGame(int[][] grid) {
        long top = 0;
        for (int i = 0; i < grid[0].length; i++) {
            top += grid[0][i];
        }
        
        long bottom = 0, res = Long.MAX_VALUE;
        for (int i = 0; i < grid[0].length; i++) {
            top -= grid[0][i];
            res = Math.min(res, Math.max(top, bottom));
            bottom += grid[1][i];
        }
        
        return res;
    }
}